integrationm/scripts/_audit.groovy

```diff
@@ -14,14 +14,10 @@ if (msg.product == "recordm-definition") cacheOfAuditFieldsForDefinition.invalid
 
 // ========================================================================================================
 def auditFields = cacheOfAuditFieldsForDefinition.get(msg.type, { getAuditFields(msg.type) })
-if (auditFields.size() > 0
-    && msg.product == "recordm"
-    && msg.user != "integrationm"
-    && msg.action =~ "add|update" ) {
-
-    def updates = getAuditFieldsUpdates(auditFields,msg.instance.fields)
-    def result = actionPacks.recordm.update(msg.type, "recordmInstanceId:" + msg.instance.id, updates);
-    if(updates) log.info("[\$audit] UPDATE '${msg.type}' id:${msg.instance.id}, updates: ${updates}, result:${result.getStatus()} | ${result.getStatusInfo()} ");
+if (auditFields.size() > 0 && msg.product == "recordm" && msg.action =~ "add|update") {
+    if (msg.user != "integrationm" || auditFields.every { f -> !msg.field(f.name).changed() }) {
```

Comment:
não percebo porque é que estás à espera de alterações nos campos de audit, o que despoleta o seu preenchimento serão alterações na instância, quaisquer que sejam, não?

Reply:
> não percebo porque é que estás à espera de alterações nos campos de audit, o que despoleta o seu preenchimento serão alterações na instância, quaisquer que sejam, não?

No, actually the script updates the audit fields whenever there is a save on the instance independent of having or not changes on any field. We want to avoid updating the audit fields when the only changes are in the audit fields.

Reply:
mas não é isso que estás a fazer, estás só recalcular se nenhum dos campos de audit mudou...não fala nada dos outros campos

Reply:
> mas não é isso que estás a fazer, estás só recalcular se nenhum dos campos de audit mudou...não fala nada dos outros campos

Está implicito, se é um add ou update algo mudou. Se não foram os próprios quer dizer que foram os outros.